lacework/resource_lacework_vulnerability_exception_host.go

```diff
@@ -0,0 +1,402 @@
+
+	vulnException := api.NewVulnerabilityException(d.Get("name").(string),
+		api.VulnerabilityExceptionConfig{
+			Description:     d.Get("description").(string),
+			Type:            api.VulnerabilityExceptionTypeHost,
+			ExceptionReason: api.NewVulnerabilityExceptionReason(d.Get("reason").(string)),
+			Severities:      api.NewVulnerabilityExceptionSeverities(severities),
+			Cve:             castAttributeToStringSlice(d, "vulnerability_criteria.0.cves"),
+			Package:         api.NewVulnerabilityExceptionPackages(packages),
+			Fixable:         d.Get("vulnerability_criteria.0.fixable").(bool),
+			ResourceScope: api.VulnerabilityExceptionHostResourceScope{
+				Hostname:    castAttributeToStringSlice(d, "resource_scope.0.hostnames"),
+				ExternalIP:  castAttributeToStringSlice(d, "resource_scope.0.external_ips"),
+				ClusterName: castAttributeToStringSlice(d, "resource_scope.0.cluster_names"),
+				Namespace:   castAttributeToStringSlice(d, "resource_scope.0.namespaces"),
+			},
+			ExpiryTime: expiry.UTC(),
+		},
+	)
+
+	if !d.Get("enabled").(bool) {
+		vulnException.Enabled = 0
+	}
+
+	log.Printf("[INFO] Creating vulnerability exception with data:\n%+v\n", vulnException)
+	response, err := lacework.V2.VulnerabilityExceptions.Create(vulnException)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(response.Data.Guid)
+	d.Set("name", response.Data.ExceptionName)
+	d.Set("guid", response.Data.Guid)
+	d.Set("enabled", response.Data.Enabled == 1)
+	d.Set("created_time", response.Data.CreatedTime)
+	d.Set("updated_time", response.Data.UpdatedTime)
+	d.Set("created_by", response.Data.Props.CreatedBy)
+	d.Set("updated_by", response.Data.Props.UpdatedBy)
+	d.Set("type", response.Data.ExceptionType)
+
+	log.Printf("[INFO] Created vulnerability exception with guid %s\n", response.Data.Guid)
+	return nil
+}
+
+func resourceLaceworkVulnerabilityExceptionHostRead(d *schema.ResourceData, meta interface{}) error {
+	var (
+		lacework = meta.(*api.Client)
+		response api.VulnerabilityExceptionResponse
+	)
+
+	log.Printf("[INFO] Reading vulnerability exception with guid %s\n", d.Id())
+	err := lacework.V2.VulnerabilityExceptions.Get(d.Id(), &response)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(response.Data.Guid)
+	d.Set("name", response.Data.ExceptionName)
+	d.Set("guid", response.Data.Guid)
+	d.Set("description", response.Data.Props.Description)
+	d.Set("enabled", response.Data.Enabled == 1)
+	d.Set("created_time", response.Data.CreatedTime)
+	d.Set("updated_time", response.Data.UpdatedTime)
+	d.Set("created_by", response.Data.Props.CreatedBy)
+	d.Set("updated_by", response.Data.Props.UpdatedBy)
+	d.Set("type", response.Data.ExceptionType)
+	d.Set("vulnerability_criteria.0.severities", response.Data.VulnerabilityCriteria.Severity)
+	d.Set("vulnerability_criteria.0.cves", response.Data.VulnerabilityCriteria.Cve)
+	d.Set("vulnerability_criteria.0.package", response.Data.VulnerabilityCriteria.Package)
```

Comment:
I am not 100% sure this set will work. that is why I want to check by adding tests to it.

Reply:
Fixed